### DISCUSSION TITLE
How do I avoid all types of mutations that occur in all types of 'loops'?

### DISCUSSION THREAD

**Discussion by realodix** [infection/infection#1527] [Q&A] [answered]
How do I avoid all types of mutations that occur in the 'while' or other types of 'loops'?

Right now, all I can do is shrink the timeout.

```diff
+ while (self::char_at($result, $start) === ' ') {
- while (self::char_at($result, $start) !== ' ') {
     $start++;
  }
```

**Comment by sanmai** [comment, marked as answer]
This should do it:
https://infection.github.io/guide/how-to.html#Do-not-mutate-the-source-code-matched-by-regular-expression

**Reply by maks-rafalko** [reply]
Agree, this is the most flexible way for now.

@realodix, please have a look at this example: https://infection-php.dev/r/372e
